packages/flutter_tools/gradle/src/main/groovy/app_plugin_loader.groovy

```diff
@@ -0,0 +1,38 @@
+import groovy.json.JsonSlurper
+import org.gradle.api.Plugin
+import org.gradle.api.initialization.Settings
+
+apply plugin: FlutterAppPluginLoaderPlugin
+
+class FlutterAppPluginLoaderPlugin implements Plugin<Settings> {
+    @Override
+    void apply(Settings settings) {
+        def flutterProjectRoot = settings.settingsDir.parentFile
+
+        // If this logic is changed, also change the logic in module_plugin_loader.gradle.
+        def pluginsFile = new File(flutterProjectRoot, '.flutter-plugins-dependencies')
+        if (!pluginsFile.exists()) {
+            return
+        }
+
+        def object = new JsonSlurper().parseText(pluginsFile.text)
+        assert object instanceof Map
+        assert object.plugins instanceof Map
+        assert object.plugins.android instanceof List
+        // Includes the Flutter plugins that support the Android platform.
+        object.plugins.android.each { androidPlugin ->
+            assert androidPlugin.name instanceof String
+            assert androidPlugin.path instanceof String
+            // Skip plugins that have no native build (such as a Dart-only implementation
```

Comment:
Please add a test that confirms that such plugins are skipped.

Reply:
Good idea - will definitely do it after I get tests to pass.

Reply:
can you resolve this when you add this test?

Reply:
I'm not sure where I should get started with writing such test.

I searched and found devicelab test called [`plugin_test`](https://github.com/flutter/flutter/blob/master/dev/devicelab/bin/tasks/plugin_test.dart), and run it locally - it passed.


```
../../bin/cache/dart-sdk/bin/dart bin/test_runner.dart test -t plugin_test
```

I commented out the snippet that @johnmccutchan suggests to write tests for. I did:

```groovy
//def needsBuild = androidPlugin.containsKey(nativeBuildKey) ? //androidPlugin[nativeBuildKey] : true
//if (!needsBuild) {
//    return
//}
```

and then run `plugin_test` again, and it failed consistently across 3 executions.

<details><summary>crash logs</summary>

```
[2023-09-14 22:23:59.607828] [STDOUT] stderr: [   +1 ms] FAILURE: Build failed with an exception.
[2023-09-14 22:23:59.607867] [STDOUT] stderr: [   +1 ms] * What went wrong:
[2023-09-14 22:23:59.607914] [STDOUT] stderr: [        ] assert pluginDirectory.exists()
[2023-09-14 22:23:59.607925] [STDOUT] stderr: [        ]        |               |
[2023-09-14 22:23:59.607934] [STDOUT] stderr: [        ]        |               false
[2023-09-14 22:23:59.607943] [STDOUT] stderr: [        ]        /private/var/folders/z8/vl6s_16515s6h_j8z7bs1tmm0000gn/T/flutter_devicelab_plugin_test.3X0VcH/plugintest/android
[2023-09-14 22:23:59.607953] [STDOUT] stderr: [        ] * Try:
[2023-09-14 22:23:59.607989] [STDOUT] stderr: [        ] > Run with --debug option to get more log output.
[2023-09-14 22:23:59.608168] [STDOUT] stderr: [        ] > Run with --scan to get full insights.
[2023-09-14 22:23:59.608185] [STDOUT] stderr: [        ] * Exception is:
[2023-09-14 22:23:59.608195] [STDOUT] stderr: [        ] Assertion failed:
[2023-09-14 22:23:59.608219] [STDOUT] stderr: [        ] assert pluginDirectory.exists()
[2023-09-14 22:23:59.608233] [STDOUT] stderr: [        ]        |               |
[2023-09-14 22:23:59.608242] [STDOUT] stderr: [        ]        |               false
[2023-09-14 22:23:59.608398] [STDOUT] stderr: [        ]        /private/var/folders/z8/vl6s_16515s6h_j8z7bs1tmm0000gn/T/flutter_devicelab_plugin_test.3X0VcH/plugintest/android
```

</details> 


So I think it's not necessary to add such test in this PR, esp. that this code is not mine – I just copied it into a new location.

Reply:
The test failing with the code commented out is good enough for me.